File: Java/Contains Duplicate.java
import java.util.HashSet;

public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(n)
     */
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> uniq = new HashSet<Integer>();

        for (int num : nums) {
            if (uniq.contains(num)) {
                return true;
            }
            uniq.add(num);
        }

        return false;
    }
}

File: Python/Contains Duplicate.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(n)
    """

    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)

        return False
